.changeset/hungry-ducks-wonder.md

```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/cli-hydrogen': patch
+---
+
+New `--debug` flag for the `dev` command that attaches a Node inspector to the development server.
```

Comment:
I don't mind this route, but I feel like it would be a better long-term solution if we could just pass any extra flags we get to the remix cli, and that way we're not reimplementing flags that already exist in remix.

Thoughts? @blittle

Reply:
We are not using the Remix CLI directly, but its internals.
Unfortunately, the usage of `--debug` in Remix CLI is in a very early point in their code that we can't reuse, that's why I'm just replicating that functionality here.
I agree that ideally we'd just reuse their whole CLI, but there are some limitations in that area (integration with MiniOxygen, custom logs, etc).

Reply:
>We are not using the Remix CLI directly, but its internals.

Ah, that's unfortunate. Thanks for the context.